### DISCUSSION TITLE
tkinter into apk

### DISCUSSION THREAD

**Discussion by isaque1309** [TomSchimansky/CustomTkinter#2166] [General]
Is it possible to transform a tkinter app into android apk?

**Comment by dipeshSam** [comment]
> Is it possible to transform a tkinter app into android apk?

The basic `Tkinter` code can be run on Android using an application [PyDroid 3](https://play.google.com/store/apps/details?id=ru.iiec.pydroid3). But you can't create standalone executable for Android i.e., .apk file. This happens because PyDroid provides necessary dependencies to the Tkinter library.

**PyDroid basic Tkinter code and compilation:**
![IMG_20231227_093528.jpg](https://github.com/TomSchimansky/CustomTkinter/assets/72375763/2340a3bd-7117-456d-ad7e-1ada75a053cb) ![IMG_20231227_093549.jpg](https://github.com/TomSchimansky/CustomTkinter/assets/72375763/08bd481d-4fb0-4f8b-a8f1-bf416fcd5a77)


**Interestingly, you can not even run the `customtkinter` on Android. These both libraries are natively built for desktop applications and not for mobile apps.**

Hope, it will be helpful for you.
Happy customtkinter :)

**Reply by isaque1309** [reply]
thank u so much for your reply :)

**Comment by Anonymous6598** [comment]
It's impossible because python's tkinter framework is not supported on android. Even kivi framework, which was designed for cross platform apps, has some setbacks. Like it's hard to recompile the script into apk. You will need google utilities for it (not 100% sure).
